internal/controller/etcd/reconcile_status.go

```diff
@@ -27,11 +27,20 @@ func (r *Reconciler) reconcileStatus(ctx component.OperatorContext, etcdObjectKe
 	}
 	sLog := r.logger.WithValues("etcd", etcdObjectKey, "operation", "reconcileStatus").WithValues("runID", ctx.RunID)
 	originalEtcd := etcd.DeepCopy()
-	mutateETCDStatusStepFns := []mutateEtcdStatusFn{
-		r.mutateETCDStatusWithMemberStatusAndConditions,
-		r.inspectStatefulSetAndMutateETCDStatus,
-		r.setSelector,
+
+	var mutateETCDStatusStepFns []mutateEtcdStatusFn
+	if druidv1alpha1.IsEtcdRuntimeComponentCreationEnabled(etcd.ObjectMeta) {
+		mutateETCDStatusStepFns = []mutateEtcdStatusFn{
+			r.mutateETCDStatusWithMemberStatusAndConditions,
+			r.inspectStatefulSetAndMutateETCDStatus,
+		}
+	} else {
+		sLog.Info("Skipping status checks since etcd runtime component creation is disabled")
```

Comment:
There is one problem I see @shreyas-s-rao ,
suppose a user deployed the Etcd CR without this annotation with `.spec.replicas: 3`, after that when all resources get created successfully then user adds this annotation `druid.gardener.cloud/disable-etcd-runtime-component-creation` (may be by mistake), now the problem what I see is druid controller will always skip the update of Etcd status but resources are already got created, IMO, this is not the correct behaviour. wdyt ?

Reply:
Agreed, thanks for pointing this out!

Reply:
@ishan16696 I've added the cleanup logic. PTAL.